src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -204,11 +206,31 @@ private void CacheInstanceDiscoveryMetadata(InstanceDiscoveryMetadataEntry metad
 
         private async Task<Uri> BuildAuthorityWithRegionAsync(Uri canonicalAuthority, RequestContext requestContext)
         {
+            string regionToUse = requestContext.ServiceBundle.Config.AuthorityInfo.RegionToUse;
+
             if (s_region.IsNullOrEmpty())
             {
-                s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+                try
+                {
+                    s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+
+                    if (!regionToUse.IsNullOrEmpty())
+                    {
```

Comment:
i would think the customer would know the region they want to target, and we should just use the provided value. what does ests suggest we do or have they provided input on that?

Reply:
why do we care if the region they provided is valid or not? like you mentioned earlier, they could provide a garbage region and then we are logging all these errors, but we have no telemetry dashboard set up to get the data on this. Why does ests care if the region provided by the customer matches the region we discover? seems like a test they should have on their side, but not necessarily something we need in MSAL.

---
In reply to: [551518052](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2315#discussion_r551518052) [](ancestors = 551518052)

Reply:
It is highly likely that people use a value for region that is different from the actual azure region and with allowing user to provide the region we are also open to the possibility of someone hardcoding this value and hitting the wrong regional endpoint. Happy to discuss this offline.

---
In reply to: [551519474](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2315#discussion_r551519474) [](ancestors = 551519474,551518052)

Reply:
highly likely or highly unlikely? if highly likely, then sounds like we should do something else to detect this sooner. also, customers won't have the telemetry logs to see the azure regions, is this also getting logged in the regular logs as well? I guess, how will they know which region they are hitting? +1 to discussing. maybe at stand-up tomorrow.

---
In reply to: [551576965](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2315#discussion_r551576965) [](ancestors = 551576965,551519474,551518052)

Reply:
Do we want to have an API on the IConfidentialClientApplication which would be string GetAzureRegion() ?